packages/compass-query-bar/src/components/query-history/query-item/query-item-action-buttons.tsx

```diff
@@ -1,54 +1,85 @@
-import { Icon, IconButton } from '@mongodb-js/compass-components';
+import { Icon, IconButton, css } from '@mongodb-js/compass-components';
 import React from 'react';
 
 type ActionButtonProps = {
-  onClick: () => void;
+  onClick?: () => void;
+  visible?: boolean;
 };
 
-export const FavoriteActionButton = ({ onClick }: ActionButtonProps) => {
+const VISIBLE_DISPLAY = 'inline';
+const INVISIBLE_DISPLAY = 'none';
+
+const exhaustIfInteractive =
```

Comment:
Why do we need that? How can onClick be optional on a button that should cause an action?

Reply:
Ah, okay, I see that it's because of https://github.com/mongodb-js/compass/pull/5105#discussion_r1391111472 and https://github.com/mongodb-js/compass/pull/5105#discussion_r1391094629. I think if you change it to use the conditional rendering and being explicit about what behavior action button will trigger, this might remove the need for that here

Reply:
Yeah, I did the changes you suggested and simplified, I'll ask you for another review when I'm done with the rest of the comments 😄